src/overwrite.js

```diff
@@ -13,25 +13,28 @@ module.exports = function overwriteCheckFactory(options) {
 };
 
 function checkFilesExist(options) {
-	const client = knox.createClient(options.getCreds());
+
+	const creds = options.getCreds();
+	const client = new AWS.S3({
+		accessKeyId: creds.key,
+		apiVersion: '2006-03-01',
+		secretAccessKey: creds.secret
+	});
 
 	return new Promise((resolve, reject) => {
-		client.list({
-			prefix: options.getUploadPath()
-		}, (err, data) => {
+		const params = {
+			Bucket: creds.bucket,
+			MaxKeys: 1,
```

Comment:
I limited it to return just 1 object since this is just checking if there's anything in the directory -- we don't care how many objects are in here.